pyproject.toml

```diff
@@ -160,6 +161,7 @@ src = ["starcraft", "tests"]
 extend-exclude = [
     "docs",
     "__pycache__",
+    "_version.py",
```

Comment:
Because it's generated by setuptools.  First two lines of the file in one of my checkouts:

```
# file generated by setuptools_scm
# don't change, don't track in version control
```

Reply:
Because the generated `_version.py` file sets off all sorts of flags for ruff and we don't actually want to include it in the repo anyway.

Reply:
Interesting, isn't `ruff` following `.gitignore` ?
I can see in the `tox.ini` we are using `--respect-gitignore`. Maybe that's the missing piece ?

Reply:
Ahh yes, that's the stuff!

Actually looking at it, the [latest ruff](https://docs.astral.sh/ruff/settings/#respect-gitignore) respects gitignore by default. @mattculler did you add this in response to ruff doing a weird, or was this pre-emptive?

Reply:
Turns out ruff does exclude files from .gitignore by default.  Removed that flag from `tox.ini` and removed duplicated ignores from `pyproject.toml`.